Catch exceptions when creating shortcuts

feerrenrut opened the pull request:
<!--
Please fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples
-->

### Link to issue number:
fixes #5166
fixes #6326

Related, alternative approach: #10874

### Summary of the issue:
The way that shortcuts are created (wsh) does not handle unicode characters. This results in the installer presenting an error. 

### Description of how this pull request fixes the issue:
Wrap each call to create shortcut in a try block, on error try to create the shortcut without translation.

### Testing performed:
Manually replaced translated versions of the strings with unicode characters (including the hotkey).
Checked that an error is presented in the log, but the shortcut is still created.

### Known issues with pull request:
- When an error occurs the shortcuts / hotkey will not be translated.

### Change log entry:

Section: New features, Changes, Bug fixes

feerrenrut committed:
Catch exceptions when creating shortcuts

Wrap each call to create shortcut in a try block, on error
try to create the shortcut without translation.

michaelDCurran commented on source/installer.py:
```diff
@@ -249,30 +249,113 @@ def registerInstallation(installDir,startMenuFolder,shouldCreateDesktopShortcut,
+	registerAddonFileAssociation(slaveExe)
+
+
+def _updateShotcuts(NVDAExe, installDir, shouldCreateDesktopShortcut, slaveExe, startMenuFolder) -> None:
```
Rename _updateShotcuts to _updateShortcuts please.

michaelDCurran reviewed (changes requested):
Lots of flake8 errors at the moment also.

feerrenrut committed:
Fix function name

feerrenrut committed:
Extract and generalise the fallback logic when creating shortcuts.

Flake8 was complaining about the code complexity, which I agreed with
and could not bring myself to make flake8 ignore the function.

feerrenrut commented:
Since the linter was complaining about the complexity of the function, and I agree it was hard to read, I have re-worked it. The new function wraps the calls to create shortcut, falling back on the non-translated name / hotkey as necessary.
